src/mouse.js

```diff
@@ -52,49 +53,68 @@ var mouseEvents = {
     e.pointerId = this.POINTER_ID;
     e.isPrimary = true;
     e.pointerType = this.POINTER_TYPE;
-    if (!HAS_BUTTONS) {
-      e.buttons = WHICH_TO_BUTTONS[e.which] || 0;
-    }
     return e;
   },
+  prepareButtonsForMove: function(e, inEvent) {
+    var p = pointermap.get(this.POINTER_ID);
+    e.buttons = p ? p.buttons : 0;
+    inEvent.buttons = e.buttons
+  },
   mousedown: function(inEvent) {
     if (!this.isEventSimulatedFromTouch(inEvent)) {
-      var p = pointermap.has(this.POINTER_ID);
+      var p = pointermap.get(this.POINTER_ID);
       // TODO(dfreedman) workaround for some elements not sending mouseup
       // http://crbug/149091
-      if (p) {
-        this.cancel(inEvent);
-      }
       var e = this.prepareEvent(inEvent);
+      if (!HAS_BUTTONS) {
+        e.buttons = BUTTON_TO_BUTTONS[e.button];
+        if (p) e.buttons |= p.buttons;
+        inEvent.buttons = e.buttons
+      }
       pointermap.set(this.POINTER_ID, inEvent);
-      dispatcher.down(e);
+      if(!p)
+        dispatcher.down(e);
+      else
+        dispatcher.move(e);
     }
   },
   mousemove: function(inEvent) {
     if (!this.isEventSimulatedFromTouch(inEvent)) {
       var e = this.prepareEvent(inEvent);
+      if (!HAS_BUTTONS) this.prepareButtonsForMove(e, inEvent);
       dispatcher.move(e);
     }
   },
   mouseup: function(inEvent) {
     if (!this.isEventSimulatedFromTouch(inEvent)) {
       var p = pointermap.get(this.POINTER_ID);
-      if (p && p.button === inEvent.button) {
-        var e = this.prepareEvent(inEvent);
-        dispatcher.up(e);
+      if (!p) return;
+      var e = this.prepareEvent(inEvent);
+      if (!HAS_BUTTONS) {
+        var up = BUTTON_TO_BUTTONS[e.button];
+        e.buttons = p.buttons & ~up;
```

Comment:
I believe on a normal left click this causes `buttons=0` for the pointerup event (Chrome 41.0). Is this intended?